src/deploy/functions/backend.ts

```diff
@@ -537,10 +560,31 @@ export async function checkAvailability(context: Context, want: Backend): Promis
   }
 }
 
-// To be a bit more deterministic, print function lists in a prescribed order.
-// Future versions might want to compare regions by GCF/Run pricing tier before
-// location.
-export function compareFunctions(left: FunctionSpec, right: FunctionSpec): number {
+/** A helper utility for flattening all endpoints in a backend since typing is a bit wonky. */
+export function allEndpoints(backend: Backend): Endpoint[] {
```

Comment:
Consider adding some light unit tests for these new methods. I wouldn't normally suggest doing so for helper methods, but these seem like utility methods that are used in a few places (and could easily become more widely used over time).